### ISSUE TITLE
Snippet does not honor the level in which it is used

### ISSUE THREAD

**Issue by deep9** [mondeja/mkdocs-include-markdown-plugin#7]
I need to use one snippet file like this

snippet.md
```
- item
- item
```

```
# What can be done
{% include-markdown "snippet.md" %}
```

and 
```
1. Is that all
    {% include-markdown "snippet-md" %}
```

While in the first the result is as expected, the second usage will create
```
<ol>
<li>
<p> Is that all </p>
<p>
- item
- item
</p>
</li>
```
instead of expected
```
<ol>
<li>
<p> Is that all </p>
<ul>
<li> item</li>
<li> item</li>
</ul>
</li>
```

**Comment by mondeja** [comment]
Sorry but I can't reproduce your problem. Have you checked if there are other plugins influencing the rendering result?

**Comment by deep9** [comment]
@mondeja Thank you for looking into this. The current setup produces following output
```
## Snippet test

You will see list

{% include-markdown "../snippet.md" %}

## OL UL mix test

1. Snippet, no indent
{% include-markdown "../snippet.md" %}
   
2. Snippet, indent 4 spaces, clse
    {% include-markdown "../snippet.md" %}
   
3. Snippet, one blank space

   {% include-markdown "../snippet.md" %}
```

produces
```
<p>You will see list</p>
<!-- BEGIN INCLUDE ../snippet.md   -->

<ul>
<li>Item 1</li>
<li>Item 2</li>
</ul>
<!-- END INCLUDE -->

<h2 id="ol-ul-mix-test">OL UL mix test<a class="headerlink" href="#ol-ul-mix-test" title="Permanent link">&para;</a></h2>
<ol>
<li>Snippet, no indent
<!-- BEGIN INCLUDE ../snippet.md   --></li>
<li>Item 1</li>
<li>Item 2</li>
</ol>
<!-- END INCLUDE -->

<ol>
<li>Snippet, indent 4 spaces, clse
    <!-- BEGIN INCLUDE ../snippet.md   --></li>
<li>Item 1</li>
<li>Item 2</li>
</ol>
<!-- END INCLUDE -->

<ol>
<li>Snippet, one blank space</li>
</ol>
<p><!-- BEGIN INCLUDE ../snippet.md   -->
- Item 1
- Item 2</p>
<!-- END INCLUDE -->
```

I will try to disable some plugins and see if the output will be any better.

**Comment by mondeja** [comment]
With the third example I'm getting this:

```html
<ol>
  <li>Snippet, one blank space</li>
</ol>
<ul>
  <li>item</li>
  <li>item</li>
</ul>
```

So probably other plugins are influencing the result. Try changing the order of the plugins in your configuration.
I'm closing this. If you provide a minimal reproducible project I will take a look, but without knowing more about your complete configuration I can't do anything.

**Comment by deep9** [comment]
@mondeja I've disabled all other plugins and extensions. I might be doing something wrong.

```
## Snippet test

You will see list

{% include-markdown "../snippet.md" %}

## OL UL mix test

1. Snippet, no indent
{% include-markdown "../snippet.md" %}
2. Snippet, indent 4 spaces, close
    {% include-markdown "../snippet.md" %}
3. Snippet, indent 4 spaces, one blank line
   
    {% include-markdown "../snippet.md" %}
4. Snippet, indent 8 spaces, close
      {% include-markdown "../snippet.md" %}

## Direct data

1. Snippet, no indent
- Item 1
- Item 2
2. Snippet, indent 4 spaces, close
    - Item 1
    - Item 2
3. Snippet, indent 4 spaces, one blank line
   
    - Item 1
    - Item 2
4. Snippet, indent 8 spaces, close
        - Item 1
        - Item 2

```

produces

```
<h2 id="snippet-test">Snippet test</h2>
<p>You will see list</p>
<!-- BEGIN INCLUDE ../snippet.md   -->

<ul>
<li>Item 1</li>
<li>Item 2</li>
</ul>
<!-- END INCLUDE -->

<h2 id="ol-ul-mix-test">OL UL mix test</h2>
<ol>
<li>Snippet, no indent
<!-- BEGIN INCLUDE ../snippet.md   --></li>
<li>Item 1</li>
<li>Item 2</li>
</ol>
<!-- END INCLUDE -->

<ol>
<li>Snippet, indent 4 spaces, close
    <!-- BEGIN INCLUDE ../snippet.md   --></li>
<li>Item 1</li>
<li>Item 2</li>
</ol>
<!-- END INCLUDE -->

<ol>
<li>
<p>Snippet, indent 4 spaces, one blank line</p>
<p><!-- BEGIN INCLUDE ../snippet.md   -->
- Item 1
- Item 2</p>
</li>
</ol>
<!-- END INCLUDE -->

<ol>
<li>Snippet, indent 8 spaces, close
      <!-- BEGIN INCLUDE ../snippet.md   --></li>
<li>Item 1</li>
<li>Item 2</li>
</ol>
<!-- END INCLUDE -->

<h2 id="direct-data">Direct data</h2>
<ol>
<li>Snippet, no indent</li>
<li>Item 1</li>
<li>Item 2</li>
<li>Snippet, indent 4 spaces, close<ul>
<li>Item 1</li>
<li>Item 2</li>
</ul>
</li>
<li>
<p>Snippet, indent 4 spaces, one blank line</p>
<ul>
<li>Item 1</li>
<li>Item 2</li>
<li>Snippet, indent 8 spaces, close<ul>
<li>Item 1</li>
<li>Item 2</li>
</ul>
</li>
</ul>
</li>
</ol>
```

![image](https://user-images.githubusercontent.com/1571715/102863531-000ffa00-4433-11eb-88e5-9313f38a4e0e.png)

**Comment by deep9** [comment]
```bash
$ pip list
Package                        Version
------------------------------ ----------
asn1crypto                     0.24.0
click                          7.1.2
cryptography                   2.6.1
entrypoints                    0.3
future                         0.18.2
importlib-metadata             3.3.0
Jinja2                         2.11.2
joblib                         1.0.0
keyring                        17.1.1
keyrings.alt                   3.1.1
livereload                     2.6.3
lunr                           0.5.8
Markdown                       3.2.2
markdown-captions              2
MarkupSafe                     1.1.1
mkdocs                         1.1.2
mkdocs-include-markdown-plugin 2.1.2
mkdocs-redirects               1.0.1
nltk                           3.5
pip                            20.3.3
pycrypto                       2.6.1
PyGObject                      3.30.4
pymdown-extensions             8.0.1
pyxdg                          0.25
PyYAML                         5.3.1
regex                          2020.11.13
SecretStorage                  2.3.1
setuptools                     40.8.0
six                            1.12.0
tornado                        5.1.1
tqdm                           4.54.1
typing-extensions              3.7.4.3
virtualenv                     15.1.0
wheel                          0.32.3
zipp                           3.4.0

```

mkdocs.yml
```
plugins:
    - include-markdown
    - search

markdown_extensions:

nav:
```

**Comment by mondeja** [comment]
```
3. Snippet, indent 4 spaces, one blank line
   
    - Item 1
    - Item 2
```
The "Item X" part is indented by 4 spaces separed by 1 line from the previous ordered list item, so it renders as an indented code block, but you don't have that markdown extension enabled, so it's rendered as a paragraph. Nothing wrong here, it's the expected behaviour.

**Comment by deep9** [comment]
@mondeja Thanks for the feedback and sorry for bothering you, I don't know what extension should I enable to make include work the same.

The image is from one file, and in my expectations the result should be the with or without the include.

Can you please point me in the right direction, how to re-use the same snippet in sub-list and on top level?

**Comment by mondeja** [comment]
I'm getting the right result using this configuration:

### Includer file

```markdown
1. Ordered list item
  {% include-markdown "../snippet.md" %}
```

### File to include (`snippet.md`)

```markdown
    - item
    - item
```

### Result

```html
<ol>
  <li>
    <p>Ordered list item <!-- BEGIN INCLUDE ../test.md   --></p>
    <ul>
      <li>item</li>
      <li>item</li>
    </ul>
  </li>
</ol>
<!-- END INCLUDE -->
```

![image](https://user-images.githubusercontent.com/23049315/102896975-98c06d00-4467-11eb-8b70-2e71a87996b2.png)

Maybe this plugin is not doing a great job with indentation, but what must be the expected behaviour? Indentation of the included file must be preserved regardless the indentation of the includer call itself? I'm not sure :thinking:
Do you have a use case where you need this configuration? Is not a bit mess including sublists from other files?

**Comment by deep9** [comment]
@mondeja There is one list
```
- item 1
- item 2
```

for navigation. It is used in several places, one of those is also in the ordered list item.

I guess I'll solve it for now by having 2 snippet files, one indented, one not.

I also have the feelint than the comment might be messing with the output, I'll play around.

Please take that indentation  preservation as a feature request, that based on where the include-markdown is used (the indentation) would propagate to all lines in the included file.

Thank you very much.

**Comment by mondeja** [comment]
Io sum this up in an approachable way. I assume that do you want is that this:

```markdown
2. Ordered list item
    {% include-markdown "../snippet.md" %}
```

Being `../snippet.md` this:

```markdown
- item
- item
```

... according to you it should produce this:

```markdown
1. Ordered list item
    - item
    - item
```

I'm right? If so, why not just use this as `../snippet.md`?

```markdown
    - item
    - item
```

If I'm not missing something, you are suggesting that the preservation of indent should happen in all lines, right? I think that this is not a good choice. Maybe this could be implemented as an option named `preserve_includer_indent`, which default value would be `false`, but implement this as a default when the unique elements that could be indent-related are lists and indented code blocks seems to me absurd from a valid markdown generation point of view. Personally, I think that the included content should not break includer file content in any way, and this change opens the door to, for example, infinite sublisting level. I think that this behaviour is counterintuitive, but could be useful to some user like you.

Anyway, you agree that this would be implemented in an expiremental option `preserve_includer_indent` which default value would be `false`? I'm asking you this because if this is what you are asking for, I will open another issue with the information summarized so that whoever needs it can implement it without having to follow the entire discussion of this issue.

**Comment by mondeja** [comment]
This is implemented since v2.3.0 as `preserve_includer_indent` option.

**Comment by deep9** [comment]
@mondeja Thank you very much for the support, you are doing a wonderful job.
